### PR TITLE
Rust 1.23 release

### PR DESCRIPTION
**Pull request by steveklabnik**

### CODE REVIEW

**Comment by CryZe** [comment]
The memory usage drop is just while compiling right? I feel like that should be specified, ad you might falsely assume that the compiled programs use less memory. (I may be wrong about that, which even more proves that this is very unclear atm)

**Comment by steveklabnik** [comment]
Uh yeah, I think you're right. @alexcrichton ?

File: _posts/2018-01-04-Rust-1.23.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.23.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1230-2018-01-04
+
+## What's in 1.23.0 stable
+
+New year, new Rust! For our first improvement today, we now [avoid some uneccesary
```
Comment (frewsxcv) [inline _posts/2018-01-04-Rust-1.23.md]: small typo: unnecessary

File: _posts/2018-01-04-Rust-1.23.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+assert!(!non_ascii.is_ascii());
+assert!(int_ascii.is_ascii());
+```
+
+In Rust 1.23, these methods are now defined directly on those types, and so you no longer need
+to import the trait. Thanks to our stability guarantees, this trait still exists, so if you'd
+like to still support Rust versions before Rust 1.23, you can do this:
+
+```rust
+#[warn(unused_imports)]
```
Comment (frewsxcv) [inline _posts/2018-01-04-Rust-1.23.md]: should this be `#[allow(unused_imports)]`?

File: _posts/2018-01-04-Rust-1.23.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+use std::ascii::AsciiExt;
+```
+
+To supress the related warning. Once you drop support for older Rusts, you
```
Comment (frewsxcv) [inline _posts/2018-01-04-Rust-1.23.md]: typo: suppress

up to you, but you could also do something like this to indicate the words after the code block are a continuation of the previous sentence:

> …to suppress the related warning. Once you drop support for older Rusts, you can remove both lines, and everything will continue to work.

File: _posts/2018-01-04-Rust-1.23.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+can remove both lines, and everything will continue to work.
+
+[`AsciiExt`]: https://doc.rust-lang.org/std/ascii/trait.AsciiExt.html
+
+Additionally, a few new APIs were stabilized this release:
+
+* The various [`std::sync::atomic
+  types`](https://doc.rust-lang.org/beta/std/sync/atomic/index.html#structs)
```
Comment (frewsxcv) [inline _posts/2018-01-04-Rust-1.23.md]: should this be linking to `/beta`?

**Comment by steveklabnik** [comment]
Thanks @frewsxcv !

**Comment by steveklabnik** [comment]
@CryZe i've fixed up the wording, you were correct!

**Pull request merged**